pkg/injector/webhook.go

```diff
@@ -89,16 +89,8 @@ func NewMutatingWebhook(config Config, kubeClient kubernetes.Interface, certMana
 	// Start the MutatingWebhook web server
 	go wh.run(stop)
 
-	if enableReconciler {
-		// Create the MutatingWebhook
-		if err = createMutatingWebhook(wh.kubeClient, webhookHandlerCert, webhookTimeout, webhookConfigName, meshName, osmNamespace, osmVersion); err != nil {
-			return errors.Errorf("Error creating MutatingWebhookConfiguration %s: %+v", webhookConfigName, err)
-		}
-	} else {
-		// Update the MutatingWebhookConfig with the OSM CA bundle only, as the MutatingWebhook is created via Helm
-		if err = updateMutatingWebhookCABundle(webhookHandlerCert, webhookConfigName, wh.kubeClient); err != nil {
-			return errors.Errorf("Error configuring MutatingWebhookConfiguration %s: %+v", webhookConfigName, err)
-		}
+	if err = createMutatingWebhook(wh.kubeClient, webhookHandlerCert, webhookTimeout, webhookConfigName, meshName, osmNamespace, osmVersion, enableReconciler); err != nil {
```

Comment:
what happens if there is a mutating webhook config from a previous version of the control plane? or if the osm-controller or another osm component is restarted?

Reply:
please check the newly added code in webhook.go, it has additional logic to handle all the scenarios

Reply:
I am adding comments to make it more readable as well, will update the PR in a bit